build/source/driver/multi_driver.f90

```diff
@@ -233,23 +234,60 @@ program multi_driver
 ! output control 
 integer(i4b)                     :: iFreq                      ! index for looping through output files
 
+! parallelize the model run
+integer(i4b)                     :: startGRU                   ! index of the starting GRU for parallelization run
+integer(i4b)                     :: startHRU                   ! index of the HRU for a single HRU run
+character(len=256),allocatable   :: argString(:)               ! string to store command line arguments
+integer(i4b),parameter           :: defaultArguments=2         ! number of command arguments for a full run
+integer(i4b),parameter           :: hruParallelization=3       ! number of command arguments for a single-HRU run
+integer(i4b),parameter           :: gruParallelization=4       ! number of command arguments for a  run
+integer(i4b),parameter           :: iArgumentFileSuffix=1      ! index of the argument of output_fileSuffix
+integer(i4b),parameter           :: iArgumentFileMan=2         ! index of the argument of summaFileManagerFile
+integer(i4b),parameter           :: iArgumentStartGRU=3        ! index of the argument of startGRU
+integer(i4b),parameter           :: iArgumentCountGRU=4        ! index of the argument of countGRU
+integer(i4b),parameter           :: iArgumentStartHRU=3        ! index of the argument of startHRU
 ! *****************************************************************************
 ! (1) inital priming -- get command line arguments, identify files, etc.
 ! *****************************************************************************
-print*, 'start'
 ! get the initial time
-call date_and_time(cdate1,ctime1)
-print*,ctime1
+call date_and_time(values=ctime1)
+print "(A,I2.2,':',I2.2,':',I2.2)", 'start at ',ctime1(5:7)
+! check numbers of command-line arguments
+select case (getCommandArguments(argString))
```

Comment:
This seems to be getting overly dependent on assumptions.

Could we clarify things by using switches.
For example:
```
summa.exe --name <name> --gruStart <gruStart> --gruCount <gruCount> --hruIndex <hruIndex>
```
Then you can check that the number of command line arguments is an even number, and process the switches using `case` statements.